x/liquiditypool/keeper/store_position.go

```diff
@@ -85,6 +85,14 @@ func (k Keeper) RemovePosition(ctx context.Context, id uint64) {
 	storeAdapter := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
 	store := prefix.NewStore(storeAdapter, types.KeyPrefix(types.PositionKey))
 	store.Delete(GetPositionIDBytes(id))
+
+	// Remove from PositionsByPoolId
+	position, found := k.GetPosition(ctx, id)
```

Comment:
If this function is executed after `store.Delete(GetPositionIDBytes(id))`, it would always return `[Empty], false`.

Shouldn't it be like below?

```go
func (k Keeper) RemovePosition(ctx context.Context, id uint64) {
	storeAdapter := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
	store := prefix.NewStore(storeAdapter, types.KeyPrefix(types.PositionKey))
	store.Delete(GetPositionIDBytes(id))

	k.RemovePositionsByPool(ctx, id)
	k.RemovePositionsByAddress(ctx, id)
}
```

```go
// Remove PositionBy Pool
func RemovePositionsByPool(...) {
  store = prefix.NewStore(storeAdapter, types.PositionByPoolPrefix(poolId))
  iterator := storetypes.KVStorePrefixIterator(store, []byte{})
  defer iterator.Close()

  for ; iterator.Valid(); iterator.Next() {
    id := sdk.BigEndianToUint64(iterator.Value())
    store.Delete(iterator.Value())
  }
}
// Remove PositionsByAddress
func RemovePositionsByAddress(...) {
  // Do similar things
}
```